pkg/asset/cluster/azure/azure.go

```diff
@@ -12,3 +20,27 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 		Region: config.Platform.Azure.Region,
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) error {
+	clusterName := installConfig.Config.ObjectMeta.Name
+	record := fmt.Sprintf("api.%s", clusterName)
+
+	azureDNS, err := azureconfig.NewDNSConfig()
+	if err != nil {
+		return err
+	}
+
+	rgName := installConfig.Config.Azure.BaseDomainResourceGroupName
+	zoneName := installConfig.Config.BaseDomain
+
+	// Look for an existing CNAME first
+	rs, err := azureDNS.GetDNSRecordSet(rgName, zoneName, record, azdns.CNAME)
+	if err == nil && rs.CnameRecord != nil {
+		// If a record exists, it's likely a cluster already exists
+		return errors.New(fmt.Sprintf("api.%s record already exists in %s and might be in use by another cluster, please remove it to continue", zoneName, clusterName))
+	}
+
```

Comment:
why did you drop the A record and AAAA record check?

Reply:
I misunderstood this comment: https://github.com/openshift/installer/pull/3120#discussion_r421012777 . I'll fix accordingly.

Reply:
Fixed.